sdk/js-connect/core/base/src/utils/metaprogramming.ts

```diff
@@ -0,0 +1,59 @@
+export type Extends<T, U> = [T] extends [U] ? true : false;
+
+export type Function<P extends any[] = any[], R = any> = (...args: P) => R;
```

Comment:
it does https://www.typescriptlang.org/docs/handbook/2/functions.html#function

cc @nonergodic do we need this definition?

Reply:
As the documentation says:
> This is an untyped function call and is generally best avoided because of the unsafe any return type.
I.e. another thing that TypeScript has and says "oh, it's there but, you shouldn't really use it. Think of it like a rake that we put in tall grass so you can accidentally step on it and have it smack you in the face. We put it there because we thought it was funny."

Ultimately, both `Function` and `Extends`, just like e.g. `RoArrayf, are simply there to make stuff more readable (in my book - reasonable people can disagree here I think :man_shrugging: )